--- abiword.orig/src/Makefile	2013-09-01 04:58:46.534849475 +0000
+++ abiword-2.8.2/src/Makefile	2013-09-01 06:24:03.026627153 +0000
@@ -500,7 +500,7 @@
 	$(WP_CPPFLAGS) \
 	-DABIWORD_DATADIR="\"$(ABIWORD_DATADIR)\""
 
-libabiword_2_8_la_LDFLAGS = $(DEPS_LIBS) --no-undefined -avoid-version \
+libabiword_2_8_la_LDFLAGS = $(DEPS_LIBS) -avoid-version \
 	-export-dynamic $(am__append_6)
 libabiword_2_8_la_LIBADD = $(opt_libs) plugins/libplugins.la \
 	text/fmt/xp/libxp.la text/ptbl/xp/libxp.la \
@@ -535,7 +535,6 @@
 
 #AbiWord_LDFLAGS = \
 #	$(DEPS_LIBS) \
-#	--no-undefined \
 #	-avoid-version \
 #	-export-dynamic \
 #	-headerpad_max_install_names 
@@ -551,7 +550,6 @@
 
 abiword_LDFLAGS = \
 	$(platform_ldflags) \
-	--no-undefined      \
 	-avoid-version      \
 	-export-dynamic
 
