--- alsa-lib-1.0.27.2/src/pcm/pcm.c	2013-07-08 12:31:36.000000000 +0000
+++ alsa-lib-1.0.27.2.patched/src/pcm/pcm.c	2013-08-05 06:49:05.000000000 +0000
@@ -7125,7 +7125,9 @@
 
 #ifndef DOC_HIDDEN
 
+#undef USE_VERSIONED_SYMBOLS
 #ifdef USE_VERSIONED_SYMBOLS
+#warning USE_VERSIONED_SYMBOLS
 
 #define OBSOLETE1(name, what, new) \
   default_symbol_version(__##name, name, new); \
@@ -7190,19 +7192,15 @@
 __OLD_GET1(snd_pcm_hw_params_get_tick_time_max, unsigned int, unsigned int);
 
 #define __P_OLD_NEAR(pfx, name, ret_type) \
-ret_type pfx##name(snd_pcm_t *pcm, snd_pcm_hw_params_t *params, ret_type val) \
+ret_type pfx##name(snd_pcm_t *pcm, snd_pcm_hw_params_t *params, ret_type *val) \
 { \
-	if (INTERNAL(name)(pcm, params, &val) < 0) \
-		return 0; \
-	return (ret_type)val; \
+	return(INTERNAL(name)(pcm, params, val)); \
 }
 
 #define __P_OLD_NEAR1(pfx, name, ret_type) \
-ret_type pfx##name(snd_pcm_t *pcm, snd_pcm_hw_params_t *params, ret_type val, int *dir) \
+ret_type pfx##name(snd_pcm_t *pcm, snd_pcm_hw_params_t *params, ret_type *val, int *dir) \
 { \
-	if (INTERNAL(name)(pcm, params, &val, dir) < 0) \
-		return 0; \
-	return (ret_type)val; \
+	return (INTERNAL(name)(pcm, params, val, dir) < 0); \
 }
 
 #define __OLD_NEAR(name, ret_type) __P_OLD_NEAR(__old_, name, ret_type)
