--- busybox-1.20.2/archival/libarchive/decompress_bunzip2.c
+++ busybox-1.20.2.patched/archival/libarchive/decompress_bunzip2.c
@@ -440,7 +440,7 @@
 		   literal used is the one at the head of the mtfSymbol array.) */
 		if (runPos != 0) {
 			uint8_t tmp_byte;
-			if (dbufCount + runCnt >= dbufSize) return RETVAL_DATA_ERROR;
+			if (dbufCount + runCnt > dbufSize) return RETVAL_DATA_ERROR;
 			tmp_byte = symToByte[mtfSymbol[0]];
 			byteCount[tmp_byte] += runCnt;
 			while (--runCnt >= 0) dbuf[dbufCount++] = (uint32_t)tmp_byte;
