--- dosbox-0.74.org/src/dos/cdrom_ioctl_linux.cpp	2013-04-13 16:06:02.070000003 +0000
+++ dosbox-0.74/src/dos/cdrom_ioctl_linux.cpp	2013-04-13 16:06:42.168000002 +0000
@@ -24,7 +24,7 @@
 #if defined (LINUX)
 #include <fcntl.h>
 #include <unistd.h>
-#include <linux/cdrom.h>
+#include "linux_cdrom.h"
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
--- /dev/null	2012-12-26 22:18:45.184000007 +0000
+++ dosbox-0.74/src/dos/linux_cdrom.h	2013-04-13 16:30:39.779000003 +0000
@@ -0,0 +1,20 @@
+#ifndef LINUX_CDROM_H
+#define LINUX_CDROM_H
+
+#define CD_FRAMESIZE 2048
+#define CD_FRAMESIZE_RAW 2352
+#define CDROM_GET_MCN 0x5311
+#define CDROMREADRAW 0x5314
+
+struct cdrom_read {
+	int cdread_lba;
+	char *cdread_bufaddr;
+	int cdread_buflen;
+};
+
+struct cdrom_mcn {
+	unsigned char medium_catalog_number[14];
+};
+
+#endif
+
