--- dtach-0.9.org/vt.c
+++ dtach-0.9/vt.c.new
@@ -1521,7 +1521,7 @@
 		mvwaddch(win, srow + b->rows - 1, 0, t->cmdline->prefix);
 		whline(win, ' ', b->cols - 1);
 		if (t->cmdline->state == CMDLINE_ACTIVE) {
-			waddnwstr(win, t->cmdline->display, b->cols - 1);
+			waddnstr(win, t->cmdline->display, b->cols - 1);
 			wmove(win, srow + b->rows - 1, 1 + t->cmdline->cursor_pos);
 		} else
 			wmove(win, srow + b->rows - 1, 1);
