From f77e6cf4fd7fef49ac91d8c62b6a9a993529adb8 Mon Sep 17 00:00:00 2001
From: Carlos Garcia Campos <carlosgc@gnome.org>
Date: Fri, 17 Sep 2010 11:21:16 +0000
Subject: [pdf] Update to poppler api changes

Linearized PopplerDocument property is now boolean rather than string.
---
diff --git a/backend/pdf/ev-poppler.cc b/backend/pdf/ev-poppler.cc
index aa080e6..ced3ef7 100644
--- a/backend/pdf/ev-poppler.cc
+++ b/backend/pdf/ev-poppler.cc
@@ -722,6 +722,9 @@ pdf_document_get_info (EvDocument *document)
 	PopplerPermissions permissions;
 	EvPage *page;
 	char *metadata;
+#ifdef HAVE_POPPLER_DOCUMENT_IS_LINEARIZED
+	gboolean linearized;
+#endif
 
 	info = g_new0 (EvDocumentInfo, 1);
 
@@ -758,7 +761,11 @@ pdf_document_get_info (EvDocument *document)
 		      "producer", &(info->producer),
 		      "creation-date", &(info->creation_date),
 		      "mod-date", &(info->modified_date),
+#ifdef HAVE_POPPLER_DOCUMENT_IS_LINEARIZED
+		      "linearized", &linearized,
+#else
 		      "linearized", &(info->linearized),
+#endif
 		      "metadata", &metadata,
 		      NULL);
 
@@ -864,6 +871,10 @@ pdf_document_get_info (EvDocument *document)
 		info->security = g_strdup (_("No"));
 	}
 
+#ifdef HAVE_POPPLER_DOCUMENT_IS_LINEARIZED
+	info->linearized = linearized ? g_strdup (_("Yes")) : g_strdup (_("No"));
+#endif
+
 	return info;
 }
 
--
cgit v0.9.0.2
