--- evince-2.32.0.org/shell/ev-window.c	2013-01-08 08:57:53.927000002 +0000
+++ evince-2.32.0/shell/ev-window.c	2013-01-08 08:59:00.412000003 +0000
@@ -7091,8 +7091,8 @@
 	/* Give focus to the document view */
 	gtk_widget_grab_focus (ev_window->priv->view);
 
-	ev_window->priv->default_settings = g_settings_new (GS_SCHEMA_NAME".Default");
-	g_settings_delay (ev_window->priv->default_settings);
+	ev_window->priv->default_settings = 0; //g_settings_new (GS_SCHEMA_NAME".Default");
+	//g_settings_delay (ev_window->priv->default_settings);
 
 	/* Set it user interface params */
 	ev_window_setup_recent (ev_window);
