--- mozilla-release.org/tools/profiler/UnwinderThread2.cpp	2013-10-25 22:27:44.000000000 +0000
+++ mozilla-release/tools/profiler/UnwinderThread2.cpp	2013-10-31 02:52:32.435000745 +0000
@@ -8,6 +8,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <time.h>
+#include <sys/types.h>
 
 #ifdef MOZ_VALGRIND
 # include <valgrind/helgrind.h>
--- mozilla-release.org/tools/profiler/local_debug_info_symbolizer.cc	2013-10-25 22:27:44.000000000 +0000
+++ mozilla-release/tools/profiler/local_debug_info_symbolizer.cc	2013-10-31 02:55:40.353986956 +0000
@@ -3,6 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include <sys/types.h>
 #include "PlatformMacros.h"
 #include "nsAutoPtr.h"
 
--- mozilla-release.org/tools/profiler/platform-linux.cc
+++ mozilla-release/tools/profiler/platform-linux.cc
@@ -560,12 +560,14 @@
 
 void TickSample::PopulateContext(void* aContext)
 {
+#ifdef __GLIBC__
   MOZ_ASSERT(aContext);
   ucontext_t* pContext = reinterpret_cast<ucontext_t*>(aContext);
   if (!getcontext(pContext)) {
     context = pContext;
     SetSampleContext(this, aContext);
   }
+#endif
 }
 
 void OS::SleepMicro(int microseconds)
