--- fuse-2.9.3.org/lib/mount_util.c
+++ fuse-2.9.3/lib/mount_util.c
@@ -16,6 +16,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <limits.h>
+#include <paths.h>
 #ifndef __NetBSD__
 #include <mntent.h>
 #endif
--- fuse-2.9.3.org/util/fusermount.c
+++ fuse-2.9.3/util/fusermount.c
@@ -20,6 +20,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <pwd.h>
+#include <paths.h>
 #include <mntent.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
