--- glib-2.36.3.org/gio/Makefile.in
+++ glib-2.36.3/gio/Makefile.in
@@ -576,7 +576,7 @@
   done | $(am__uniquify_input)`
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = gdbus-2.0/codegen xdgmime inotify kqueue fen win32 . \
+DIST_SUBDIRS = xdgmime inotify kqueue fen win32 . \
 	fam tests
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 am__relativize = \
@@ -858,7 +858,7 @@
 	gnetworking.h.win32 $(NULL) $(completion_DATA)
 TEST_PROGS = 
 NULL = 
-SUBDIRS = gdbus-2.0/codegen $(am__append_1) $(am__append_4) \
+SUBDIRS = $(am__append_1) $(am__append_4) \
 	$(am__append_7) $(am__append_11) $(am__append_14) . \
 	$(am__append_17) $(am__append_24)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
