--- iptables-1.4.14/include/net/netfilter/nf_conntrack_tuple.h
+++ iptables-1.4.14.patched/include/net/netfilter/nf_conntrack_tuple.h
@@ -15,7 +15,6 @@
 #define _NF_CONNTRACK_TUPLE_H
 
 #include <linux/netfilter/x_tables.h>
-#include <linux/netfilter/nf_conntrack_tuple_common.h>
 
 /* A `tuple' is a structure containing the information to uniquely
   identify a connection.  ie. if two packets have the same tuple, they
