From f6b58d2bdc0a24ce94dedce59802f091979df265 Mon Sep 17 00:00:00 2001
From: rofl0r <retnyg@gmx.net>
Date: Thu, 22 Nov 2012 16:26:21 +0100
Subject: [PATCH] fix build on ppc when long double == double

---
 src/powerpc/ffi.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/powerpc/ffi.c b/src/powerpc/ffi.c
index baca694..257f02f 100644
--- a/src/powerpc/ffi.c
+++ b/src/powerpc/ffi.c
@@ -664,9 +664,11 @@ ffi_prep_cif_machdep (ffi_cif *cif)
   switch (type)
     {
 #ifndef __NO_FPRS__
+#if FFI_TYPE_LONGDOUBLE != FFI_TYPE_DOUBLE
     case FFI_TYPE_LONGDOUBLE:
       flags |= FLAG_RETURNS_128BITS;
       /* Fall through.  */
+#endif
     case FFI_TYPE_DOUBLE:
       flags |= FLAG_RETURNS_64BITS;
       /* Fall through.  */
-- 
1.7.3.4

