--- lxc-0.9.0.org/src/lxc/start.c
+++ lxc-0.9.0/src/lxc/start.c
@@ -24,7 +24,6 @@
 #include "config.h"
 
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <string.h>
 #include <stdlib.h>
 #include <dirent.h>
--- lxc-0.9.0.org/src/lxc/cgroup.c
+++ lxc-0.9.0/src/lxc/cgroup.c
@@ -22,7 +22,6 @@
  */
 #define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <errno.h>
 #include <unistd.h>
--- lxc-0.9.0.org/src/lxc/lxc_console.c
+++ lxc-0.9.0/src/lxc/lxc_console.c
@@ -23,7 +23,6 @@
 
 #define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <errno.h>
 #include <string.h>
--- lxc-0.9.0.org/src/lxc/lxc_restart.c
+++ lxc-0.9.0/src/lxc/lxc_restart.c
@@ -22,7 +22,6 @@
  */
 #define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <unistd.h>
 #include <errno.h>
--- lxc-0.9.0.org/src/lxc/restart.c
+++ lxc-0.9.0/src/lxc/restart.c
@@ -24,7 +24,6 @@
 #include "config.h"
 
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <string.h>
 #include <stdlib.h>
 #include <errno.h>
--- lxc-0.9.0.org/src/lxc/freezer.c
+++ lxc-0.9.0/src/lxc/freezer.c
@@ -22,7 +22,6 @@
  */
 #define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <errno.h>
 #include <unistd.h>
--- lxc-0.9.0.org/src/lxc/conf.c
+++ lxc-0.9.0/src/lxc/conf.c
@@ -22,7 +22,6 @@
  */
 #define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <stdarg.h>
 #include <errno.h>
--- lxc-0.9.0.org/src/lxc/lxc_start.c
+++ lxc-0.9.0/src/lxc/lxc_start.c
@@ -22,7 +22,6 @@
  */
 #define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <libgen.h>
 #include <stdlib.h>
 #include <unistd.h>
--- lxc-0.9.0.org/src/lxc/parse.c
+++ lxc-0.9.0/src/lxc/parse.c
@@ -22,7 +22,6 @@
  */
 #define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <string.h>
 #include <stdlib.h>
 #include <errno.h>
