--- mdadm-3.2.6.org/platform-intel.c	2013-02-01 20:41:53.717000003 +0000
+++ mdadm-3.2.6/platform-intel.c	2013-02-01 20:48:23.468000004 +0000
@@ -116,7 +116,7 @@
 
 		list->dev_id = (__u16) dev_id;
 		list->type = type;
-		list->path = canonicalize_file_name(path);
+		list->path = realpath(path, NULL);
 		list->next = NULL;
 		if ((list->pci_id = strrchr(list->path, '/')) != NULL)
 			list->pci_id++;
@@ -459,7 +459,7 @@
 	char device[46];
 
 	sprintf(device, "/sys/dev/block/%d:%d/device", major(dev), minor(dev));
-	return canonicalize_file_name(device);
+	return realpath(device, NULL);
 }
 
 char *diskfd_to_devpath(int fd)
