--- mdadm-3.2.6.org/mdadm.h	2013-02-01 20:41:53.711000003 +0000
+++ mdadm-3.2.6/mdadm.h	2013-02-01 20:42:40.706000003 +0000
@@ -25,13 +25,6 @@
 #define	_GNU_SOURCE
 #define _FILE_OFFSET_BITS 64
 #include	<unistd.h>
-#if !defined(__dietlibc__) && !defined(__KLIBC__)
-extern __off64_t lseek64 __P ((int __fd, __off64_t __offset, int __whence));
-#else
-# if defined(__NO_STAT64) || __WORDSIZE != 32
-# define lseek64 lseek
-# endif
-#endif
 
 #include	<sys/types.h>
 #include	<sys/stat.h>
