0.9.8 upto 2ad9cf52e
diff --git a/arch/arm/bits/mman.h b/arch/arm/bits/mman.h
index 03f961e..2f23d44 100644
--- a/arch/arm/bits/mman.h
+++ b/arch/arm/bits/mman.h
@@ -15,7 +15,11 @@
 #define MAP_FILE       0x00
 #define MAP_ANON       0x20
 #define MAP_ANONYMOUS  MAP_ANON
-#define MAP_32BIT      0x40
+#define MAP_NORESERVE  0x4000
+#define MAP_GROWSDOWN  0x0100
+#define MAP_DENYWRITE  0x0800
+#define MAP_EXECUTABLE 0x1000
+#define MAP_LOCKED     0x2000
 
 #define POSIX_MADV_NORMAL       0
 #define POSIX_MADV_RANDOM       1
diff --git a/arch/arm/bits/user.h b/arch/arm/bits/user.h
index 9df7a9a..3e5a4d2 100644
--- a/arch/arm/bits/user.h
+++ b/arch/arm/bits/user.h
@@ -26,6 +26,7 @@ struct user {
 	unsigned long u_tsize, u_dsize, u_ssize;
 	unsigned long start_code, start_stack;
 	long signal;
+	int reserved;
 	struct user_regs *u_ar0;
 	unsigned long magic;
 	char u_comm[32];
diff --git a/arch/i386/bits/mman.h b/arch/i386/bits/mman.h
index 03f961e..4083c19 100644
--- a/arch/i386/bits/mman.h
+++ b/arch/i386/bits/mman.h
@@ -16,6 +16,11 @@
 #define MAP_ANON       0x20
 #define MAP_ANONYMOUS  MAP_ANON
 #define MAP_32BIT      0x40
+#define MAP_NORESERVE  0x4000
+#define MAP_GROWSDOWN  0x0100
+#define MAP_DENYWRITE  0x0800
+#define MAP_EXECUTABLE 0x1000
+#define MAP_LOCKED     0x2000
 
 #define POSIX_MADV_NORMAL       0
 #define POSIX_MADV_RANDOM       1
diff --git a/arch/i386/bits/signal.h b/arch/i386/bits/signal.h
index 29455ae..2943756 100644
--- a/arch/i386/bits/signal.h
+++ b/arch/i386/bits/signal.h
@@ -49,6 +49,26 @@ typedef struct __ucontext {
 #define SA_RESETHAND  0x80000000
 #define SA_RESTORER   0x04000000
 
+#define REG_GS		0
+#define REG_FS		1
+#define REG_ES		2
+#define REG_DS		3
+#define REG_EDI		4
+#define REG_ESI		5
+#define REG_EBP		6
+#define REG_ESP		7
+#define REG_EBX		8
+#define REG_EDX		9
+#define REG_ECX		10
+#define REG_EAX		11
+#define REG_TRAPNO	12
+#define REG_ERR		13
+#define REG_EIP		14
+#define REG_CS		15
+#define REG_EFL		16
+#define REG_UESP	17
+#define REG_SS		18
+
 #endif
 
 #define SIGHUP    1
@@ -86,3 +106,4 @@ typedef struct __ucontext {
 #define SIGUNUSED SIGSYS
 
 #define _NSIG 65
+
diff --git a/arch/microblaze/bits/mman.h b/arch/microblaze/bits/mman.h
index 03f961e..2f23d44 100644
--- a/arch/microblaze/bits/mman.h
+++ b/arch/microblaze/bits/mman.h
@@ -15,7 +15,11 @@
 #define MAP_FILE       0x00
 #define MAP_ANON       0x20
 #define MAP_ANONYMOUS  MAP_ANON
-#define MAP_32BIT      0x40
+#define MAP_NORESERVE  0x4000
+#define MAP_GROWSDOWN  0x0100
+#define MAP_DENYWRITE  0x0800
+#define MAP_EXECUTABLE 0x1000
+#define MAP_LOCKED     0x2000
 
 #define POSIX_MADV_NORMAL       0
 #define POSIX_MADV_RANDOM       1
diff --git a/arch/mips/bits/mman.h b/arch/mips/bits/mman.h
index f40e451..3e6faa4 100644
--- a/arch/mips/bits/mman.h
+++ b/arch/mips/bits/mman.h
@@ -15,6 +15,11 @@
 #define MAP_FILE       0x00
 #define MAP_ANON       0x800
 #define MAP_ANONYMOUS  MAP_ANON
+#define MAP_NORESERVE  0x0400
+#define MAP_GROWSDOWN  0x1000
+#define MAP_DENYWRITE  0x2000
+#define MAP_EXECUTABLE 0x4000
+#define MAP_LOCKED     0x8000
 
 #define POSIX_MADV_NORMAL       0
 #define POSIX_MADV_RANDOM       1
diff --git a/arch/powerpc/bits/mman.h b/arch/powerpc/bits/mman.h
index 03f961e..0c6cc32 100644
--- a/arch/powerpc/bits/mman.h
+++ b/arch/powerpc/bits/mman.h
@@ -15,7 +15,11 @@
 #define MAP_FILE       0x00
 #define MAP_ANON       0x20
 #define MAP_ANONYMOUS  MAP_ANON
-#define MAP_32BIT      0x40
+#define MAP_NORESERVE  0x40
+#define MAP_GROWSDOWN  0x0100
+#define MAP_DENYWRITE  0x0800
+#define MAP_EXECUTABLE 0x1000
+#define MAP_LOCKED     0x80
 
 #define POSIX_MADV_NORMAL       0
 #define POSIX_MADV_RANDOM       1
diff --git a/arch/x86_64/bits/mman.h b/arch/x86_64/bits/mman.h
index 80ed39f..c4cd8ae 100644
--- a/arch/x86_64/bits/mman.h
+++ b/arch/x86_64/bits/mman.h
@@ -16,6 +16,11 @@
 #define MAP_ANON       0x20
 #define MAP_ANONYMOUS  MAP_ANON
 #define MAP_32BIT      0x40
+#define MAP_NORESERVE  0x4000
+#define MAP_GROWSDOWN  0x0100
+#define MAP_DENYWRITE  0x0800
+#define MAP_EXECUTABLE 0x1000
+#define MAP_LOCKED     0x2000
 
 #define POSIX_MADV_NORMAL       0
 #define POSIX_MADV_RANDOM       1
diff --git a/arch/x86_64/bits/signal.h b/arch/x86_64/bits/signal.h
index 9043aea..069c6c3 100644
--- a/arch/x86_64/bits/signal.h
+++ b/arch/x86_64/bits/signal.h
@@ -52,6 +52,30 @@ typedef struct __ucontext {
 #define SA_RESETHAND  0x80000000
 #define SA_RESTORER   0x04000000
 
+#define REG_R8		0
+#define REG_R9		1
+#define REG_R10		2
+#define REG_R11		3
+#define REG_R12		4
+#define REG_R13		5
+#define REG_R14		6
+#define REG_R15		7
+#define REG_RDI		8
+#define REG_RSI		9
+#define REG_RBP		10
+#define REG_RBX		11
+#define REG_RDX		12
+#define REG_RAX		13
+#define REG_RCX		14
+#define REG_RSP		15
+#define REG_RIP		16
+#define REG_EFL		17
+#define REG_CSGFS	18
+#define REG_ERR		19
+#define REG_TRAPNO	20
+#define REG_OLDMASK	21
+#define REG_CR2		22
+
 #endif
 
 #define SIGHUP    1
@@ -89,3 +113,4 @@ typedef struct __ucontext {
 #define SIGUNUSED SIGSYS
 
 #define _NSIG 65
+
diff --git a/include/dirent.h b/include/dirent.h
index 726067f..d445f80 100644
--- a/include/dirent.h
+++ b/include/dirent.h
@@ -9,7 +9,7 @@ extern "C" {
 
 #define __NEED_ino_t
 #define __NEED_off_t
-#ifdef _BSD_SOURCE
+#if defined(_BSD_SOURCE) || defined(_GNU_SOURCE)
 #define __NEED_size_t
 #endif
 
@@ -53,16 +53,13 @@ int scandir(const char *, struct dirent ***, int (*)(const struct dirent *), int
 #define DT_WHT 14
 #define IFTODT(x) ((x)>>12 & 017)
 #define DTTOIF(x) ((x)<<12)
+int getdents(int, struct dirent *, size_t);
 #endif
 
 #ifdef _GNU_SOURCE
 int versionsort(const struct dirent **, const struct dirent **);
 #endif
 
-#ifdef _BSD_SOURCE
-int getdents(int, struct dirent *, size_t);
-#endif
-
 #if defined(_LARGEFILE64_SOURCE) || defined(_GNU_SOURCE)
 #define dirent64 dirent
 #define readdir64 readdir
@@ -72,10 +69,8 @@ int getdents(int, struct dirent *, size_t);
 #define versionsort64 versionsort
 #define off64_t off_t
 #define ino64_t ino_t
-#ifdef _BSD_SOURCE
 #define getdents64 getdents
 #endif
-#endif
 
 #ifdef __cplusplus
 }
diff --git a/include/fcntl.h b/include/fcntl.h
index 70d4cbb..b797cd1 100644
--- a/include/fcntl.h
+++ b/include/fcntl.h
@@ -108,6 +108,21 @@ int posix_fallocate(int, off_t, off_t);
 #define F_TLOCK 2
 #define F_TEST  3
 
+#define F_SETLEASE	1024
+#define F_GETLEASE	1025
+#define F_NOTIFY	1026
+#define F_CANCELLK	1029
+#define F_SETPIPE_SZ	1031
+#define F_GETPIPE_SZ	1032
+
+#define DN_ACCESS	0x00000001
+#define DN_MODIFY	0x00000002
+#define DN_CREATE	0x00000004
+#define DN_DELETE	0x00000008
+#define DN_RENAME	0x00000010
+#define DN_ATTRIB	0x00000020
+#define DN_MULTISHOT	0x80000000
+
 int lockf(int, int, off_t);
 #endif
 
diff --git a/include/features.h b/include/features.h
index a7919f3..294c61d 100644
--- a/include/features.h
+++ b/include/features.h
@@ -1,6 +1,10 @@
 #ifndef _FEATURES_H
 #define _FEATURES_H
 
+#ifdef _ALL_SOURCE
+#define _GNU_SOURCE 1
+#endif
+
 #if !defined(_POSIX_SOURCE) && !defined(_POSIX_C_SOURCE) \
  && !defined(_XOPEN_SOURCE) && !defined(_GNU_SOURCE) \
  && !defined(_BSD_SOURCE) && !defined(__STRICT_ANSI__)
diff --git a/include/inttypes.h b/include/inttypes.h
index 3f0339c..05d54ba 100644
--- a/include/inttypes.h
+++ b/include/inttypes.h
@@ -24,8 +24,11 @@ uintmax_t wcstoumax(const wchar_t *__restrict, wchar_t **__restrict, int);
 
 #if !defined __cplusplus || defined __STDC_FORMAT_MACROS
 
+#if UINTPTR_MAX == UINT64_MAX
+#define __PRI64  "l"
+#else
 #define __PRI64  "ll"
-#define __PRIPTR "l"
+#endif
 
 #define PRId8  "d"
 #define PRId16 "d"
@@ -124,12 +127,12 @@ uintmax_t wcstoumax(const wchar_t *__restrict, wchar_t **__restrict, int);
 #define PRIxMAX __PRI64 "x"
 #define PRIXMAX __PRI64 "X"
 
-#define PRIdPTR __PRIPTR "d"
-#define PRIiPTR __PRIPTR "i"
-#define PRIoPTR __PRIPTR "o"
-#define PRIuPTR __PRIPTR "u"
-#define PRIxPTR __PRIPTR "x"
-#define PRIXPTR __PRIPTR "X"
+#define PRIdPTR "ld"
+#define PRIiPTR "li"
+#define PRIoPTR "lo"
+#define PRIuPTR "lu"
+#define PRIxPTR "lx"
+#define PRIXPTR "lX"
 
 #define SCNd8   "hhd"
 #define SCNd16  "hd"
@@ -142,8 +145,8 @@ uintmax_t wcstoumax(const wchar_t *__restrict, wchar_t **__restrict, int);
 #define SCNdLEAST64 __PRI64 "d"
 
 #define SCNdFAST8  "hhd"
-#define SCNdFAST16 __PRIPTR "d"
-#define SCNdFAST32 __PRIPTR "d"
+#define SCNdFAST16 "d"
+#define SCNdFAST32 "d"
 #define SCNdFAST64 __PRI64 "d"
 
 #define SCNi8   "hhi"
@@ -157,8 +160,8 @@ uintmax_t wcstoumax(const wchar_t *__restrict, wchar_t **__restrict, int);
 #define SCNiLEAST64 __PRI64 "i"
 
 #define SCNiFAST8  "hhi"
-#define SCNiFAST16 __PRIPTR "i"
-#define SCNiFAST32 __PRIPTR "i"
+#define SCNiFAST16 "i"
+#define SCNiFAST32 "i"
 #define SCNiFAST64 __PRI64 "i"
 
 #define SCNu8   "hhu"
@@ -172,8 +175,8 @@ uintmax_t wcstoumax(const wchar_t *__restrict, wchar_t **__restrict, int);
 #define SCNuLEAST64 __PRI64 "u"
 
 #define SCNuFAST8 "hhu"
-#define SCNuFAST16 __PRIPTR "u"
-#define SCNuFAST32 __PRIPTR "u"
+#define SCNuFAST16 "u"
+#define SCNuFAST32 "u"
 #define SCNuFAST64 __PRI64 "u"
 
 #define SCNo8   "hho"
@@ -187,8 +190,8 @@ uintmax_t wcstoumax(const wchar_t *__restrict, wchar_t **__restrict, int);
 #define SCNoLEAST64 __PRI64 "o"
 
 #define SCNoFAST8  "hho"
-#define SCNoFAST16 __PRIPTR "o"
-#define SCNoFAST32 __PRIPTR "o"
+#define SCNoFAST16 "o"
+#define SCNoFAST32 "o"
 #define SCNoFAST64 __PRI64 "o"
 
 #define SCNx8   "hhx"
@@ -202,8 +205,8 @@ uintmax_t wcstoumax(const wchar_t *__restrict, wchar_t **__restrict, int);
 #define SCNxLEAST64 __PRI64 "x"
 
 #define SCNxFAST8  "hhx"
-#define SCNxFAST16 __PRIPTR "x"
-#define SCNxFAST32 __PRIPTR "x"
+#define SCNxFAST16 "x"
+#define SCNxFAST32 "x"
 #define SCNxFAST64 __PRI64 "x"
 
 #define SCNdMAX __PRI64 "d"
@@ -212,11 +215,11 @@ uintmax_t wcstoumax(const wchar_t *__restrict, wchar_t **__restrict, int);
 #define SCNuMAX __PRI64 "u"
 #define SCNxMAX __PRI64 "x"
 
-#define SCNdPTR __PRIPTR "d"
-#define SCNiPTR __PRIPTR "i"
-#define SCNoPTR __PRIPTR "o"
-#define SCNuPTR __PRIPTR "u"
-#define SCNxPTR __PRIPTR "x"
+#define SCNdPTR "ld"
+#define SCNiPTR "li"
+#define SCNoPTR "lo"
+#define SCNuPTR "lu"
+#define SCNxPTR "lx"
 
 #endif
 
diff --git a/include/math.h b/include/math.h
index 9069140..b44738d 100644
--- a/include/math.h
+++ b/include/math.h
@@ -85,21 +85,36 @@ int __signbitl(long double);
 
 #define isunordered(x,y) (isnan((x)) ? ((void)(y),1) : isnan((y)))
 
-static __inline int __isrel(long double __x, long double __y, int __rel)
-{
-	if (isunordered(__x, __y)) return 0;
-	if (__rel==-2) return __x < __y;
-	if (__rel==2) return __x > __y;
-	if (__rel==-1) return __x <= __y;
-	if (__rel==1) return __x >= __y;
-	return __x != __y;
-}
-
-#define isless(x,y) __isrel((x), (y), -2)
-#define islessequal(x,y) __isrel((x), (y), -1)
-#define islessgreater(x,y) __isrel((x), (y), 0)
-#define isgreaterequal(x,y) __isrel((x), (y), 1)
-#define isgreater(x,y) __isrel((x), (y), 2)
+#define __ISREL_DEF(rel, op, type) \
+static __inline int __is##rel(type __x, type __y) \
+{ return !isunordered(__x,__y) && __x op __y; }
+
+__ISREL_DEF(lessf, <, float)
+__ISREL_DEF(less, <, double)
+__ISREL_DEF(lessl, <, long double)
+__ISREL_DEF(lessequalf, <=, float)
+__ISREL_DEF(lessequal, <=, double)
+__ISREL_DEF(lessequall, <=, long double)
+__ISREL_DEF(lessgreaterf, !=, float)
+__ISREL_DEF(lessgreater, !=, double)
+__ISREL_DEF(lessgreaterl, !=, long double)
+__ISREL_DEF(greaterf, >, float)
+__ISREL_DEF(greater, >, double)
+__ISREL_DEF(greaterl, >, long double)
+__ISREL_DEF(greaterequalf, >=, float)
+__ISREL_DEF(greaterequal, >=, double)
+__ISREL_DEF(greaterequall, >=, long double)
+
+#define __tg_pred_2(x, y, p) ( \
+	sizeof((x)+(y)) == sizeof(float) ? p##f(x, y) : \
+	sizeof((x)+(y)) == sizeof(double) ? p(x, y) : \
+	p##l(x, y) )
+
+#define isless(x, y)            __tg_pred_2(x, y, __isless)
+#define islessequal(x, y)       __tg_pred_2(x, y, __islessequal)
+#define islessgreater(x, y)     __tg_pred_2(x, y, __islessgreater)
+#define isgreater(x, y)         __tg_pred_2(x, y, __isgreater)
+#define isgreaterequal(x, y)    __tg_pred_2(x, y, __isgreaterequal)
 
 double      acos(double);
 float       acosf(float);
diff --git a/include/netinet/tcp.h b/include/netinet/tcp.h
index 6c2d289..94e54cf 100644
--- a/include/netinet/tcp.h
+++ b/include/netinet/tcp.h
@@ -18,6 +18,8 @@
 #define TCP_CONGESTION	 13
 #define TCP_MD5SIG	 14
 
+#define SOL_TCP 6
+
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 #include <sys/types.h>
 #include <sys/socket.h>
diff --git a/include/scsi/scsi.h b/include/scsi/scsi.h
new file mode 100644
index 0000000..8837f58
--- /dev/null
+++ b/include/scsi/scsi.h
@@ -0,0 +1,150 @@
+#ifndef _SCSI_SCSI_H
+#define _SCSI_SCSI_H
+
+#define TEST_UNIT_READY 0x00
+#define REZERO_UNIT 0x01
+#define REQUEST_SENSE 0x03
+#define FORMAT_UNIT 0x04
+#define READ_BLOCK_LIMITS 0x05
+#define REASSIGN_BLOCKS 0x07
+#define READ_6 0x08
+#define WRITE_6 0x0a
+#define SEEK_6 0x0b
+#define READ_REVERSE 0x0f
+#define WRITE_FILEMARKS 0x10
+#define SPACE 0x11
+#define INQUIRY 0x12
+#define RECOVER_BUFFERED_DATA 0x14
+#define MODE_SELECT 0x15
+#define RESERVE 0x16
+#define RELEASE 0x17
+#define COPY 0x18
+#define ERASE 0x19
+#define MODE_SENSE 0x1a
+#define START_STOP 0x1b
+#define RECEIVE_DIAGNOSTIC 0x1c
+#define SEND_DIAGNOSTIC 0x1d
+#define ALLOW_MEDIUM_REMOVAL 0x1e
+#define SET_WINDOW 0x24
+#define READ_CAPACITY 0x25
+#define READ_10 0x28
+#define WRITE_10 0x2a
+#define SEEK_10 0x2b
+#define WRITE_VERIFY 0x2e
+#define VERIFY 0x2f
+#define SEARCH_HIGH 0x30
+#define SEARCH_EQUAL 0x31
+#define SEARCH_LOW 0x32
+#define SET_LIMITS 0x33
+#define PRE_FETCH 0x34
+#define READ_POSITION 0x34
+#define SYNCHRONIZE_CACHE 0x35
+#define LOCK_UNLOCK_CACHE 0x36
+#define READ_DEFECT_DATA 0x37
+#define MEDIUM_SCAN 0x38
+#define COMPARE 0x39
+#define COPY_VERIFY 0x3a
+#define WRITE_BUFFER 0x3b
+#define READ_BUFFER 0x3c
+#define UPDATE_BLOCK 0x3d
+#define READ_LONG 0x3e
+#define WRITE_LONG 0x3f
+#define CHANGE_DEFINITION 0x40
+#define WRITE_SAME 0x41
+#define READ_TOC 0x43
+#define LOG_SELECT 0x4c
+#define LOG_SENSE 0x4d
+#define MODE_SELECT_10 0x55
+#define RESERVE_10 0x56
+#define RELEASE_10 0x57
+#define MODE_SENSE_10 0x5a
+#define PERSISTENT_RESERVE_IN 0x5e
+#define PERSISTENT_RESERVE_OUT 0x5f
+#define MOVE_MEDIUM 0xa5
+#define READ_12 0xa8
+#define WRITE_12 0xaa
+#define WRITE_VERIFY_12 0xae
+#define SEARCH_HIGH_12 0xb0
+#define SEARCH_EQUAL_12 0xb1
+#define SEARCH_LOW_12 0xb2
+#define READ_ELEMENT_STATUS 0xb8
+#define SEND_VOLUME_TAG 0xb6
+#define WRITE_LONG_2 0xea
+#define GOOD 0x00
+#define CHECK_CONDITION 0x01
+#define CONDITION_GOOD 0x02
+#define BUSY 0x04
+#define INTERMEDIATE_GOOD 0x08
+#define INTERMEDIATE_C_GOOD 0x0a
+#define RESERVATION_CONFLICT 0x0c
+#define COMMAND_TERMINATED 0x11
+#define QUEUE_FULL 0x14
+#define STATUS_MASK 0x3e
+#define NO_SENSE 0x00
+#define RECOVERED_ERROR 0x01
+#define NOT_READY 0x02
+#define MEDIUM_ERROR 0x03
+#define HARDWARE_ERROR 0x04
+#define ILLEGAL_REQUEST 0x05
+#define UNIT_ATTENTION 0x06
+#define DATA_PROTECT 0x07
+#define BLANK_CHECK 0x08
+#define COPY_ABORTED 0x0a
+#define ABORTED_COMMAND 0x0b
+#define VOLUME_OVERFLOW 0x0d
+#define MISCOMPARE 0x0e
+#define TYPE_DISK 0x00
+#define TYPE_TAPE 0x01
+#define TYPE_PROCESSOR 0x03
+#define TYPE_WORM 0x04
+#define TYPE_ROM 0x05
+#define TYPE_SCANNER 0x06
+#define TYPE_MOD 0x07
+#define TYPE_MEDIUM_CHANGER 0x08
+#define TYPE_ENCLOSURE 0x0d
+#define TYPE_NO_LUN 0x7f
+#define COMMAND_COMPLETE 0x00
+#define EXTENDED_MESSAGE 0x01
+#define EXTENDED_MODIFY_DATA_POINTER 0x00
+#define EXTENDED_SDTR 0x01
+#define EXTENDED_EXTENDED_IDENTIFY 0x02
+#define EXTENDED_WDTR 0x03
+#define SAVE_POINTERS 0x02
+#define RESTORE_POINTERS 0x03
+#define DISCONNECT 0x04
+#define INITIATOR_ERROR 0x05
+#define ABORT 0x06
+#define MESSAGE_REJECT 0x07
+#define NOP 0x08
+#define MSG_PARITY_ERROR 0x09
+#define LINKED_CMD_COMPLETE 0x0a
+#define LINKED_FLG_CMD_COMPLETE 0x0b
+#define BUS_DEVICE_RESET 0x0c
+#define INITIATE_RECOVERY 0x0f
+#define RELEASE_RECOVERY 0x10
+#define SIMPLE_QUEUE_TAG 0x20
+#define HEAD_OF_QUEUE_TAG 0x21
+#define ORDERED_QUEUE_TAG 0x22
+#define SCSI_IOCTL_GET_IDLUN 0x5382
+#define SCSI_IOCTL_TAGGED_ENABLE 0x5383
+#define SCSI_IOCTL_TAGGED_DISABLE 0x5384
+#define SCSI_IOCTL_PROBE_HOST 0x5385
+#define SCSI_IOCTL_GET_BUS_NUMBER 0x5386
+
+struct ccs_modesel_head {
+	unsigned char _r1;
+	unsigned char medium;
+	unsigned char _r2;
+	unsigned char block_desc_length;
+	unsigned char density;
+	unsigned char number_blocks_hi;
+	unsigned char number_blocks_med;
+	unsigned char number_blocks_lo;
+	unsigned char _r3;
+	unsigned char block_length_hi;
+	unsigned char block_length_med;
+	unsigned char block_length_lo;
+};
+
+#endif
+
diff --git a/include/scsi/sg.h b/include/scsi/sg.h
new file mode 100644
index 0000000..f3fd05b
--- /dev/null
+++ b/include/scsi/sg.h
@@ -0,0 +1,129 @@
+#ifndef _SCSI_SG_H
+#define _SCSI_SG_H
+
+#define SG_DXFER_NONE -1
+#define SG_DXFER_TO_DEV -2
+#define SG_DXFER_FROM_DEV -3
+#define SG_DXFER_TO_FROM_DEV -4
+#define SG_FLAG_DIRECT_IO 1
+#define SG_FLAG_LUN_INHIBIT 2
+#define SG_FLAG_NO_DXFER 0x10000
+#define SG_INFO_OK_MASK 0x1
+#define SG_INFO_OK 0x0
+#define SG_INFO_CHECK 0x1
+#define SG_INFO_DIRECT_IO_MASK 0x6
+#define SG_INFO_INDIRECT_IO 0x0
+#define SG_INFO_DIRECT_IO 0x2
+#define SG_INFO_MIXED_IO 0x4
+#define SG_EMULATED_HOST 0x2203
+#define SG_SET_TRANSFORM 0x2204
+#define SG_GET_TRANSFORM 0x2205
+#define SG_SET_RESERVED_SIZE 0x2275
+#define SG_GET_RESERVED_SIZE 0x2272
+#define SG_GET_SCSI_ID 0x2276
+#define SG_SET_FORCE_LOW_DMA 0x2279
+#define SG_GET_LOW_DMA 0x227a
+#define SG_SET_FORCE_PACK_ID 0x227b
+#define SG_GET_PACK_ID 0x227c
+#define SG_GET_NUM_WAITING 0x227d
+#define SG_GET_SG_TABLESIZE 0x227F
+#define SG_GET_VERSION_NUM 0x2282
+#define SG_SCSI_RESET 0x2284
+#define SG_SCSI_RESET_NOTHING 0
+#define SG_SCSI_RESET_DEVICE 1
+#define SG_SCSI_RESET_BUS 2
+#define SG_SCSI_RESET_HOST 3
+#define SG_IO 0x2285
+#define SG_GET_REQUEST_TABLE 0x2286
+#define SG_SET_KEEP_ORPHAN 0x2287
+#define SG_GET_KEEP_ORPHAN 0x2288
+#define SG_SCATTER_SZ (8 * 4096)
+#define SG_DEFAULT_RETRIES 1
+#define SG_DEF_FORCE_LOW_DMA 0
+#define SG_DEF_FORCE_PACK_ID 0
+#define SG_DEF_KEEP_ORPHAN 0
+#define SG_DEF_RESERVED_SIZE SG_SCATTER_SZ
+#define SG_MAX_QUEUE 16
+#define SG_BIG_BUFF SG_DEF_RESERVED_SIZE
+#define SG_MAX_SENSE 16
+#define SG_SET_TIMEOUT 0x2201
+#define SG_GET_TIMEOUT 0x2202
+#define SG_GET_COMMAND_Q 0x2270
+#define SG_SET_COMMAND_Q 0x2271
+#define SG_SET_DEBUG 0x227e
+#define SG_NEXT_CMD_LEN 0x2283
+#define SG_DEFAULT_TIMEOUT (60*HZ)
+#define SG_DEF_COMMAND_Q 0
+#define SG_DEF_UNDERRUN_FLAG 0
+
+typedef struct sg_iovec {
+	void *iov_base;
+	unsigned long iov_len;
+} sg_iovec_t;
+
+typedef struct sg_io_hdr { 
+	int interface_id; 
+	int dxfer_direction; 
+	unsigned char cmd_len;
+	unsigned char mx_sb_len;
+	unsigned short iovec_count;
+	unsigned dxfer_len;
+	void *dxferp;
+	unsigned char *cmdp;
+	unsigned char *sbp;
+	unsigned timeout;
+	unsigned flags;
+	int pack_id;
+	void *usr_ptr;
+	unsigned char status;
+	unsigned char masked_status;
+	unsigned char msg_status;
+	unsigned char sb_len_wr;
+	unsigned short host_status;
+	unsigned short driver_status;
+	int resid; 
+	unsigned int duration;
+	unsigned int info;
+} sg_io_hdr_t;
+
+struct sg_scsi_id {
+	int host_no;
+	int channel;
+	int scsi_id;
+	int lun;
+	int scsi_type;
+	short h_cmd_per_lun;
+	short d_queue_depth;
+	int unused[2];
+};
+
+typedef struct sg_req_info {
+	char req_state;
+	char orphan;
+	char sg_io_owned;
+	char problem;
+	int pack_id;
+	void *usr_ptr;
+	unsigned duration; 
+	int unused; 
+} sg_req_info_t;
+
+typedef struct sg_io_hdr Sg_io_hdr;
+typedef struct sg_io_vec Sg_io_vec;
+typedef struct sg_scsi_id Sg_scsi_id;
+typedef struct sg_req_info Sg_req_info;
+
+struct sg_header {
+	int pack_len;
+	int reply_len;
+	int pack_id;
+	int result;
+	unsigned twelve_byte:1;
+	unsigned target_status:5;
+	unsigned host_status:8;
+	unsigned driver_status:8;
+	unsigned other_flags:10;
+	unsigned char sense_buffer[SG_MAX_SENSE];
+};
+
+#endif
diff --git a/include/signal.h b/include/signal.h
index 860f742..b8ba1b5 100644
--- a/include/signal.h
+++ b/include/signal.h
@@ -208,7 +208,11 @@ void (*sigset(int, void (*)(int)))(int);
 #define SIGSTKSZ 8192
 #endif
 
-#ifdef _BSD_SOURCE
+#if defined(_XOPEN_SOURCE) || defined(_GNU_SOURCE)
+#define NSIG _NSIG
+#endif
+
+#if defined(_BSD_SOURCE) || defined(_GNU_SOURCE)
 typedef void (*sig_t)(int);
 #endif
 
@@ -216,9 +220,11 @@ typedef void (*sig_t)(int);
 typedef void (*sighandler_t)(int);
 void (*bsd_signal(int, void (*)(int)))(int);
 int sigisemptyset(const sigset_t *);
+int sigorset (sigset_t *, sigset_t *, sigset_t *);
+int sigandset(sigset_t *, sigset_t *, sigset_t *);
+
 #define SA_NOMASK SA_NODEFER
 #define SA_ONESHOT SA_RESETHAND
-#define NSIG _NSIG
 #endif
 
 #include <bits/signal.h>
diff --git a/include/spawn.h b/include/spawn.h
index 92b77f7..29c799e 100644
--- a/include/spawn.h
+++ b/include/spawn.h
@@ -57,8 +57,8 @@ int posix_spawnattr_getsigdefault(const posix_spawnattr_t *__restrict, sigset_t
 
 int posix_spawnattr_setschedparam(posix_spawnattr_t *__restrict, const struct sched_param *__restrict);
 int posix_spawnattr_getschedparam(const posix_spawnattr_t *__restrict, struct sched_param *__restrict);
-int posix_spawnattr_setschedpolicy(posix_spawnattr_t *__restrict, int);
-int posix_spawnattr_getschedpolicy(const posix_spawnattr_t *__restrict, int *);
+int posix_spawnattr_setschedpolicy(posix_spawnattr_t *, int);
+int posix_spawnattr_getschedpolicy(const posix_spawnattr_t *__restrict, int *__restrict);
 
 int posix_spawn_file_actions_init(posix_spawn_file_actions_t *);
 int posix_spawn_file_actions_destroy(posix_spawn_file_actions_t *);
diff --git a/include/stddef.h b/include/stddef.h
index dbf5a4a..a5bb9ec 100644
--- a/include/stddef.h
+++ b/include/stddef.h
@@ -14,6 +14,10 @@
 
 #include <bits/alltypes.h>
 
+#if __GNUC__ > 3
+#define offsetof(type, member) __builtin_offsetof(type, member)
+#else
 #define offsetof(type, member) ((size_t)( (char *)&(((type *)0)->member) - (char *)0 ))
+#endif
 
 #endif
diff --git a/include/stdio.h b/include/stdio.h
index 9a20937..6e8e645 100644
--- a/include/stdio.h
+++ b/include/stdio.h
@@ -174,9 +174,6 @@ int ferror_unlocked(FILE *);
 int fileno_unlocked(FILE *);
 int getw(FILE *);
 int putw(int, FILE *);
-#endif
-
-#ifdef _BSD_SOURCE
 char *fgetln(FILE *, size_t *);
 #endif
 
diff --git a/include/string.h b/include/string.h
index 33f0137..c2f8eb5 100644
--- a/include/string.h
+++ b/include/string.h
@@ -81,9 +81,6 @@ void *memccpy (void *__restrict, const void *__restrict, int, size_t);
 
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 char *strsep(char **, const char *);
-#endif
-
-#ifdef _BSD_SOURCE
 size_t strlcat (char *, const char *, size_t);
 size_t strlcpy (char *, const char *, size_t);
 #endif
diff --git a/include/sys/ipc.h b/include/sys/ipc.h
index 3f896b8..f4cb9ac 100644
--- a/include/sys/ipc.h
+++ b/include/sys/ipc.h
@@ -13,10 +13,8 @@ extern "C" {
 
 #include <bits/alltypes.h>
 
-#ifdef _GNU_SOURCE
-#define __ipc_perm_key key
-#define __ipc_perm_seq seq
-#endif
+#define __ipc_perm_key __key
+#define __ipc_perm_seq __seq
 
 #include <bits/ipc.h>
 
diff --git a/include/sys/msg.h b/include/sys/msg.h
index 460275c..ceedd1c 100644
--- a/include/sys/msg.h
+++ b/include/sys/msg.h
@@ -37,6 +37,13 @@ int msgget (key_t, int);
 ssize_t msgrcv (int, void *, size_t, long, int);
 int msgsnd (int, const void *, size_t, int);
 
+#if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
+struct msgbuf {
+	long mtype;
+	char mtext[1];
+};
+#endif
+
 #ifdef __cplusplus
 }
 #endif
diff --git a/include/sys/personality.h b/include/sys/personality.h
new file mode 100644
index 0000000..852c024
--- /dev/null
+++ b/include/sys/personality.h
@@ -0,0 +1,6 @@
+#ifndef _PERSONALITY_H
+#define _PERSONALITY_H
+
+int personality(unsigned long);
+
+#endif
diff --git a/include/sys/stat.h b/include/sys/stat.h
index c63c6b8..c6abab5 100644
--- a/include/sys/stat.h
+++ b/include/sys/stat.h
@@ -88,11 +88,8 @@ int mkfifoat(int, const char *, mode_t);
 int futimens(int, const struct timespec [2]);
 int utimensat(int, const char *, const struct timespec [2], int);
 
-#ifdef _BSD_SOURCE
-int lchmod(const char *, mode_t);
-#endif
-
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
+int lchmod(const char *, mode_t);
 #define S_IREAD S_IRUSR
 #define S_IWRITE S_IWUSR
 #define S_IEXEC S_IXUSR
diff --git a/include/sys/time.h b/include/sys/time.h
index a0ed8e0..559e817 100644
--- a/include/sys/time.h
+++ b/include/sys/time.h
@@ -31,6 +31,7 @@ int utimes (const char *, const struct timeval [2]);
 
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 int futimes(int, const struct timeval [2]);
+int futimesat(int, const char *, const struct timeval [2]);
 int lutimes(const char *, const struct timeval [2]);
 int settimeofday (const struct timeval *, void *);
 int adjtime (const struct timeval *, struct timeval *);
diff --git a/include/unistd.h b/include/unistd.h
index 2791c3d..64a6b46 100644
--- a/include/unistd.h
+++ b/include/unistd.h
@@ -168,7 +168,7 @@ int getdtablesize(void);
 int sethostname(const char *, size_t);
 int getdomainname(char *, size_t);
 int setdomainname(const char *, size_t);
-int setgroups(size_t, const gid_t []);
+int setgroups(size_t, const gid_t *);
 char *getpass(const char *);
 int daemon(int, int);
 void setusershell(void);
diff --git a/src/aio/lio_listio.c b/src/aio/lio_listio.c
index 53f9f50..532b17c 100644
--- a/src/aio/lio_listio.c
+++ b/src/aio/lio_listio.c
@@ -67,7 +67,7 @@ static void *wait_thread(void *p)
 	return 0;
 }
 
-int lio_listio(int mode, struct aiocb *restrict const cbs[restrict], int cnt, struct sigevent *restrict sev)
+int lio_listio(int mode, struct aiocb *restrict const *restrict cbs, int cnt, struct sigevent *restrict sev)
 {
 	int i, ret;
 	struct lio_state *st=0;
diff --git a/src/env/__libc_start_main.c b/src/env/__libc_start_main.c
index ee10b0d..04a454e 100644
--- a/src/env/__libc_start_main.c
+++ b/src/env/__libc_start_main.c
@@ -3,6 +3,7 @@
 
 void __init_tls(size_t *);
 void __init_security(size_t *);
+void __init_ldso_ctors(void);
 
 #define AUX_CNT 38
 
@@ -37,6 +38,10 @@ int __libc_start_main(
 	/* Execute constructors (static) linked into the application */
 	if (init) init(argc, argv, envp);
 
+#ifdef SHARED
+	__init_ldso_ctors();
+#endif
+
 	/* Pass control to to application */
 	exit(main(argc, argv, envp));
 	return 0;
diff --git a/src/ldso/dynlink.c b/src/ldso/dynlink.c
index c436b88..d564b8f 100644
--- a/src/ldso/dynlink.c
+++ b/src/ldso/dynlink.c
@@ -999,13 +999,16 @@ void *__dynlink(int argc, char **argv)
 
 	if (ssp_used) __init_ssp((void *)aux[AT_RANDOM]);
 
-	atexit(do_fini);
-	do_init_fini(tail);
-
 	errno = 0;
 	return (void *)aux[AT_ENTRY];
 }
 
+void __init_ldso_ctors(void)
+{
+	atexit(do_fini);
+	do_init_fini(tail);
+}
+
 void *dlopen(const char *file, int mode)
 {
 	struct dso *volatile p, *orig_tail, *next;
diff --git a/src/misc/arch_prctl.c b/src/misc/arch_prctl.c
new file mode 100644
index 0000000..2868d66
--- /dev/null
+++ b/src/misc/arch_prctl.c
@@ -0,0 +1,9 @@
+#include "syscall.h"
+#ifdef SYS_arch_prctl
+#include "libc.h"
+int __arch_prctl(int code, unsigned long addr)
+{
+	return syscall(SYS_arch_prctl, code, addr);	
+}
+weak_alias(__arch_prctl, arch_prctl);
+#endif
diff --git a/src/misc/personality.c b/src/misc/personality.c
new file mode 100644
index 0000000..6e13a77
--- /dev/null
+++ b/src/misc/personality.c
@@ -0,0 +1,7 @@
+#include "syscall.h"
+#ifdef SYS_personality
+int personality(unsigned long persona)
+{
+	return syscall(SYS_personality, persona);	
+}
+#endif
diff --git a/src/setjmp/arm/longjmp.s b/src/setjmp/arm/longjmp.s
index 84ddc22..aff15fb 100644
--- a/src/setjmp/arm/longjmp.s
+++ b/src/setjmp/arm/longjmp.s
@@ -21,8 +21,6 @@ longjmp:
 2:	tst r1,#0x40
 	beq 2f
 	ldc p11, cr8, [ip], #64
-	ldmia ip!, {r2,r3}
-	mcr p10, 7, r3, cr1, cr0, 0
 2:	tst r1,#0x200
 	beq 3f
 	ldcl p1, cr10, [ip], #8
diff --git a/src/setjmp/arm/setjmp.s b/src/setjmp/arm/setjmp.s
index 904ff10..b74dfc6 100644
--- a/src/setjmp/arm/setjmp.s
+++ b/src/setjmp/arm/setjmp.s
@@ -23,8 +23,6 @@ setjmp:
 2:	tst r1,#0x40
 	beq 2f
 	stc p11, cr8, [ip], #64
-	mrc p10, 7, r2, cr1, cr0, 0
-	stmia ip!, {r0,r2}
 2:	tst r1,#0x200
 	beq 3f
 	stcl p1, cr10, [ip], #8
diff --git a/src/setjmp/mips/longjmp.s b/src/setjmp/mips/longjmp.s
index fc8e726..a972d67 100644
--- a/src/setjmp/mips/longjmp.s
+++ b/src/setjmp/mips/longjmp.s
@@ -10,9 +10,7 @@ longjmp:
 	bne     $2, $0, 1f
 	nop
 	addu    $2, $2, 1
-1:	lw      $8,  48($4)
-	ctc1    $8,  $31
-	lwc1    $20, 56($4)
+1:	lwc1    $20, 56($4)
 	lwc1    $21, 60($4)
 	lwc1    $22, 64($4)
 	lwc1    $23, 68($4)
diff --git a/src/setjmp/mips/setjmp.s b/src/setjmp/mips/setjmp.s
index 1b79525..53d702a 100644
--- a/src/setjmp/mips/setjmp.s
+++ b/src/setjmp/mips/setjmp.s
@@ -21,8 +21,6 @@ setjmp:
 	sw      $23, 36($4)
 	sw      $30, 40($4)
 	sw      $28, 44($4)
-	cfc1    $8, $31
-	sw      $8,  48($4)
 	swc1    $20, 56($4)
 	swc1    $21, 60($4)
 	swc1    $22, 64($4)
diff --git a/src/signal/sigandset.c b/src/signal/sigandset.c
new file mode 100644
index 0000000..e0c6f48
--- /dev/null
+++ b/src/signal/sigandset.c
@@ -0,0 +1,12 @@
+#define _GNU_SOURCE
+#include <signal.h>
+
+#define SST_SIZE (_NSIG/8/sizeof(long))
+
+int sigandset(sigset_t *dest, sigset_t *left, sigset_t *right)
+{
+	unsigned long i = 0, *d = (void*) dest, *l = (void*) left, *r = (void*) right;
+	for(; i < SST_SIZE; i++) d[i] = l[i] & r[i];
+	return 0;
+}
+
diff --git a/src/signal/sigorset.c b/src/signal/sigorset.c
new file mode 100644
index 0000000..df1b1b1
--- /dev/null
+++ b/src/signal/sigorset.c
@@ -0,0 +1,12 @@
+#define _GNU_SOURCE
+#include <signal.h>
+
+#define SST_SIZE (_NSIG/8/sizeof(long))
+
+int sigorset(sigset_t *dest, sigset_t *left, sigset_t *right)
+{
+	unsigned long i = 0, *d = (void*) dest, *l = (void*) left, *r = (void*) right;
+	for(; i < SST_SIZE; i++) d[i] = l[i] | r[i];
+	return 0;
+}
+
diff --git a/src/stat/futimesat.c b/src/stat/futimesat.c
new file mode 100644
index 0000000..0cc1854
--- /dev/null
+++ b/src/stat/futimesat.c
@@ -0,0 +1,9 @@
+#include <sys/time.h>
+#include "syscall.h"
+
+#ifdef SYS_futimesat
+int futimesat(int dirfd, const char *pathname, const struct timeval times[2])
+{
+	return syscall(SYS_futimesat, dirfd, pathname, times);
+}
+#endif
diff --git a/src/thread/pthread_attr_get.c b/src/thread/pthread_attr_get.c
index e4650e4..ad913c5 100644
--- a/src/thread/pthread_attr_get.c
+++ b/src/thread/pthread_attr_get.c
@@ -11,7 +11,7 @@ int pthread_attr_getguardsize(const pthread_attr_t *restrict a, size_t *restrict
 	return 0;
 }
 
-int pthread_attr_getinheritsched(const pthread_attr_t *a, int *inherit)
+int pthread_attr_getinheritsched(const pthread_attr_t *restrict a, int *restrict inherit)
 {
 	*inherit = a->_a_sched;
 	return 0;
@@ -23,7 +23,7 @@ int pthread_attr_getschedparam(const pthread_attr_t *restrict a, struct sched_pa
 	return 0;
 }
 
-int pthread_attr_getschedpolicy(const pthread_attr_t *a, int *policy)
+int pthread_attr_getschedpolicy(const pthread_attr_t *restrict a, int *restrict policy)
 {
 	*policy = a->_a_policy;
 	return 0;
-- 
1.7.3.4

diff --git a/src/malloc/aligned_alloc.c b/src/malloc/aligned_alloc.c
index d623420..158dba4 100644
--- a/src/malloc/aligned_alloc.c
+++ b/src/malloc/aligned_alloc.c
@@ -31,8 +31,6 @@ void *aligned_alloc(size_t align, size_t len)
 		return NULL;
 
 	header = ((size_t *)mem)[-1];
-	end = mem + (header & -8);
-	footer = ((size_t *)end)[-2];
 	new = (void *)((uintptr_t)mem + align-1 & -align);
 
 	if (!(header & 7)) {
@@ -41,6 +39,9 @@ void *aligned_alloc(size_t align, size_t len)
 		return new;
 	}
 
+	end = mem + (header & -8);
+	footer = ((size_t *)end)[-2];
+
 	((size_t *)mem)[-1] = header&7 | new-mem;
 	((size_t *)new)[-2] = footer&7 | new-mem;
 	((size_t *)new)[-1] = header&7 | end-new;
-- 
1.7.3.4

