diff --git a/src/misc/getopt_long.c b/src/misc/getopt_long.c
index ffbcd42..b4ce3c7 100644
--- a/src/misc/getopt_long.c
+++ b/src/misc/getopt_long.c
@@ -23,6 +23,7 @@ static int __getopt_long_core(int argc, char *const *argv, const char *optstring
 static int __getopt_long(int argc, char *const *argv, const char *optstring, const struct option *longopts, int *idx, int longonly)
 {
 	int ret, skipped, resumed;
+	optarg = 0;
 	if (!optind || __optreset) {
 		__optreset = 0;
 		__optpos = 0;
