diff --git a/arch/x86_64/bits/socket.h b/arch/x86_64/bits/socket.h
index 878ab11..c37ae1e 100644
--- a/arch/x86_64/bits/socket.h
+++ b/arch/x86_64/bits/socket.h
@@ -1,10 +1,10 @@
 struct msghdr
 {
 	void *msg_name;
-	socklen_t msg_namelen;
+	size_t msg_namelen;
 	struct iovec *msg_iov;
-	int msg_iovlen, __pad1;
+	int msg_iovlen;
 	void *msg_control;
-	socklen_t msg_controllen, __pad2;
+	size_t msg_controllen;
 	int msg_flags;
 };
diff --git a/include/sys/socket.h b/include/sys/socket.h
index 50de321..4c9af7c 100644
--- a/include/sys/socket.h
+++ b/include/sys/socket.h
@@ -19,7 +19,7 @@ extern "C" {
 
 struct cmsghdr
 {
-	socklen_t cmsg_len;
+	size_t cmsg_len;
 	int cmsg_level;
 	int cmsg_type;
 };
diff --git a/src/network/recvmsg.c b/src/network/recvmsg.c
index 4f52665..aeea154 100644
--- a/src/network/recvmsg.c
+++ b/src/network/recvmsg.c
@@ -6,17 +6,6 @@
 ssize_t recvmsg(int fd, struct msghdr *msg, int flags)
 {
 	ssize_t r;
-#if LONG_MAX > INT_MAX
-	struct msghdr h, *orig = msg;
-	if (msg) {
-		h = *msg;
-		h.__pad1 = h.__pad2 = 0;
-		msg = &h;
-	}
-#endif
 	r = socketcall_cp(recvmsg, fd, msg, flags, 0, 0, 0);
-#if LONG_MAX > INT_MAX
-	if (orig) *orig = h;
-#endif
 	return r;
 }
diff --git a/src/network/sendmsg.c b/src/network/sendmsg.c
index 164c28d..52a6982 100644
--- a/src/network/sendmsg.c
+++ b/src/network/sendmsg.c
@@ -5,13 +5,5 @@
 
 ssize_t sendmsg(int fd, const struct msghdr *msg, int flags)
 {
-#if LONG_MAX > INT_MAX
-	struct msghdr h;
-	if (msg) {
-		h = *msg;
-		h.__pad1 = h.__pad2 = 0;
-		msg = &h;
-	}
-#endif
 	return socketcall_cp(sendmsg, fd, msg, flags, 0, 0, 0);
 }
