From 2b44fb783c29aee47bae7bdecf67912f7890830f Mon Sep 17 00:00:00 2001
From: rofl0r <retnyg@gmx.net>
Date: Sat, 14 Jul 2012 17:47:07 +0200
Subject: [PATCH] fix getservbyname

---
 src/network/getservbyname_r.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/network/getservbyname_r.c b/src/network/getservbyname_r.c
index 5c02515..9615fd4 100644
--- a/src/network/getservbyname_r.c
+++ b/src/network/getservbyname_r.c
@@ -22,9 +22,13 @@ int getservbyname_r(const char *name, const char *prots,
 	buf += sizeof(char *)-i;
 	buflen -= sizeof(char *)-i;
 
-	if (!strcmp(prots, "tcp")) hint.ai_protocol = IPPROTO_TCP;
+	if(!prots) hint.ai_protocol = 0;
+	else if (!strcmp(prots, "tcp")) hint.ai_protocol = IPPROTO_TCP;
 	else if (!strcmp(prots, "udp")) hint.ai_protocol = IPPROTO_UDP;
-	else return -1;
+	else {
+		errno = EINVAL;
+		return -1;
+	}
 
 	if (getaddrinfo(0, name, &hint, &ai) < 0) return -1;
 
-- 
1.7.3.4

