diff --git a/src/mman/mmap.c b/src/mman/mmap.c
index fd2bb07..0b27eb2 100644
--- a/src/mman/mmap.c
+++ b/src/mman/mmap.c
@@ -14,8 +14,10 @@ void *__mmap(void *start, size_t len, int prot, int flags, int fd, off_t off)
 {
 	void *ret;
 	if (sizeof(off_t) > sizeof(long))
-		if (((long)off & 0xfff) | ((long)((unsigned long long)off>>(12 + 8*(sizeof(off_t)-sizeof(long))))))
-			start = (void *)-1;
+		if (((long)off & 0xfff) | ((long)((unsigned long long)off>>(12 + 8*(sizeof(off_t)-sizeof(long)))))) {
+			errno = EINVAL;
+			return MAP_FAILED;
+		}
 	if (flags & MAP_FIXED) __vm_lock(-1);
 #ifdef SYS_mmap2
 	ret = (void *)syscall(SYS_mmap2, start, len, prot, flags, fd, off>>12);
-- 
1.7.3.4

