Subject: [PATCH] Makefile: fix regression causing DESTDIR to appear in the ldso symlink
diff --git a/Makefile b/Makefile
index aa5aee8..7ac58d4 100644
--- a/Makefile
+++ b/Makefile
@@ -157,7 +157,7 @@ $(DESTDIR)$(includedir)/%: include/%
 	$(INSTALL) -D -m 644 $< $@
 
 $(DESTDIR)$(LDSO_PATHNAME): $(DESTDIR)$(libdir)/libc.so
-	$(INSTALL) -D -l $< $@ || true
+	$(INSTALL) -D -l $(libdir)/libc.so $@ || true
 
 install-libs: $(ALL_LIBS:lib/%=$(DESTDIR)$(libdir)/%) $(if $(SHARED_LIBS),$(DESTDIR)$(LDSO_PATHNAME),)
 
-- 
1.7.3.4

