--- netsurf-all-3.2.org/libsvgtiny/src/colors.gperf	2014-09-05 16:58:04.266781938 +0200
+++ netsurf-all-3.2/libsvgtiny/src/colors.gperf	2014-09-05 16:58:32.636780584 +0200
@@ -21,10 +21,8 @@
  * generated function from being inlined.  This is pointless given
  * it (a) is in a separate .c file and (b) has external linkage.
  */
-#ifdef __inline
 #undef __inline
-#define __inline
-#endif
+#define __inline()
 %}
 
 struct svgtiny_named_color;
