--- qt-x11-opensource-src-4.3.4.org/src/corelib/codecs/qiconvcodec.cpp	2013-06-29 19:57:32.017000004 +0000
+++ qt-x11-opensource-src-4.3.4/src/corelib/codecs/qiconvcodec.cpp	2013-06-30 01:48:57.339000004 +0000
@@ -62,8 +62,15 @@
 #elif defined(Q_OS_AIX)
 #  define NO_BOM
 #  define UTF16 "UCS-2"
-#else
+#elif defined(__GLIBC__)
 #  define UTF16 "UTF-16"
+#else
+#  define NO_BOM
+#  if Q_BYTE_ORDER == Q_BIG_ENDIAN
+#    define UTF16 "UTF-16BE"
+#  else
+#    define UTF16 "UTF-16LE"
+#  endif
 #endif
 
 #ifdef Q_OS_MAC
