--- qt-x11-opensource-src-4.3.4.org/qmake/property.cpp	2013-06-29 19:57:32.500000004 +0000
+++ qt-x11-opensource-src-4.3.4/qmake/property.cpp	2013-06-30 03:00:27.151000004 +0000
@@ -182,6 +181,25 @@
                     fprintf(stdout, "%s:%s\n", (*it2).toLatin1().constData(), ret.toLatin1().constData());
                 }
             }
+            QStringList specialProps;
+            specialProps.append("QT_INSTALL_PREFIX");
+            specialProps.append("QT_INSTALL_DATA");
+            specialProps.append("QT_INSTALL_DOCS");
+            specialProps.append("QT_INSTALL_HEADERS");
+            specialProps.append("QT_INSTALL_LIBS");
+            specialProps.append("QT_INSTALL_BINS");
+            specialProps.append("QT_INSTALL_PLUGINS");
+            specialProps.append("QT_INSTALL_TRANSLATIONS");
+            specialProps.append("QT_INSTALL_CONFIGURATION");
+            specialProps.append("QT_INSTALL_EXAMPLES");
+            specialProps.append("QT_INSTALL_DEMOS");
+            specialProps.append("QMAKE_MKSPECS");
+            specialProps.append("QMAKE_VERSION");
+#ifdef QT_VERSION_STR
+            specialProps.append("QT_VERSION");
+#endif
+            foreach (QString prop, specialProps)
+                fprintf(stdout, "%s:%s\n", prop.toLatin1().constData(), value(prop).toLatin1().constData());
             return true;
         }
         for(QStringList::ConstIterator it = Option::prop::properties.begin();
