--- qt-x11-opensource-src-4.3.4.org/src/network/qsslcertificate.cpp
+++ qt-x11-opensource-src-4.3.4/src/network/qsslcertificate.cpp
@@ -368,7 +368,7 @@
 {
     QMultiMap<QSsl::AlternateNameEntryType, QString> result;
 
-    STACK *altNames = (STACK *)q_X509_get_ext_d2i(d->x509, NID_subject_alt_name, 0, 0);
+    STACK_OF(GENERAL_NAME) *altNames = (STACK_OF(GENERAL_NAME) *)q_X509_get_ext_d2i(d->x509, NID_subject_alt_name, 0, 0);
 
     if (altNames) {
         for (int i = 0; i < q_sk_GENERAL_NAME_num(altNames); ++i) {
@@ -380,7 +380,7 @@
             else if (genName->type == GEN_EMAIL)
                 result.insert(QSsl::EmailEntry, altName);
         }
-        q_sk_free(altNames);
+        q_sk_free((STACK*)altNames);
     }
 
     return result;
--- qt-x11-opensource-src-4.3.4.org/src/network/qsslsocket_openssl_p.h
+++ qt-x11-opensource-src-4.3.4/src/network/qsslsocket_openssl_p.h
@@ -87,6 +87,7 @@
 #include <openssl/rand.h>
 #include <openssl/ssl.h>
 #include <openssl/stack.h>
+#define STACK _STACK
 #include <openssl/x509.h>
 #include <openssl/x509v3.h>
 #include <openssl/x509_vfy.h>
