--- qt-everywhere-opensource-src-4.6.4.org/src/3rdparty/clucene/src/CLucene/util/Misc.cpp	2013-10-25 12:36:39.159114738 +0000
+++ qt-everywhere-opensource-src-4.6.4/src/3rdparty/clucene/src/CLucene/util/Misc.cpp	2013-10-25 19:20:45.228244399 +0000
@@ -20,6 +20,7 @@
 # endif
 #endif
 
+#undef _CL_HAVE_SYS_TIMEB_H
 #ifdef _CL_HAVE_SYS_TIMEB_H
 # include <sys/timeb.h>
 #endif
--- qt-everywhere-opensource-src-4.6.4.org/src/3rdparty/javascriptcore/JavaScriptCore/wtf/RefPtr.h	2013-10-25 12:36:38.840114749 +0000
+++ qt-everywhere-opensource-src-4.6.4/src/3rdparty/javascriptcore/JavaScriptCore/wtf/RefPtr.h	2013-10-25 19:18:15.880249760 +0000
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include "AlwaysInline.h"
 #include "FastAllocBase.h"
+#include "PassRefPtr.h"
 
 namespace WTF {
 
