--- qt-everywhere-opensource-src-4.6.4.org/tools/linguist/linguist/mainwindow.cpp	2013-10-25 12:36:39.796114715 +0000
+++ qt-everywhere-opensource-src-4.6.4/tools/linguist/linguist/mainwindow.cpp	2013-10-25 19:36:14.497211042 +0000
@@ -2370,7 +2370,7 @@
             // because we get a lot of false positives.
             if (c != '&' && c != ' ' && QChar(c).isPrint()) {
                 const ushort *pp = p;
-                for (; *p < 256 && ::isalpha(*p); p++) ;
+                for (; *p < 256 && isalpha(*p); p++) ;
                 if (pp == p || *p != ';')
                     return true;
                 // This looks like a HTML &entity;, so ignore it. As a HTML string
