commit a31aa886f29ab49bb0aa757ce2253aa4e8ef3632
Author: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@nokia.com>
Date:   Tue Mar 16 13:43:02 2010 +0100

    Compile
    
    Fix typo in qabstractitemview.cpp
    
    Reviewed-by: Thierry

diff --git a/src/gui/itemviews/qabstractitemview.cpp b/src/gui/itemviews/qabstractitemview.cpp
index fb6e418..dd9a7e6 100644
--- a/src/gui/itemviews/qabstractitemview.cpp
+++ b/src/gui/itemviews/qabstractitemview.cpp
@@ -2298,7 +2298,7 @@ void QAbstractItemView::keyPressEvent(QKeyEvent *event)
     case Qt::Key_Shift:
     case Qt::Key_Control:
     case Qt::Key_Delete:
-    case Qt::Key_BackSpace:
+    case Qt::Key_Backspace:
         event->ignore();
         break;
     case Qt::Key_Space:
