--- qt-everywhere-opensource-src-4.6.4.org/src/corelib/io/qurl.h	2013-10-25 12:36:38.698114754 +0000
+++ qt-everywhere-opensource-src-4.6.4/src/corelib/io/qurl.h	2013-10-25 16:55:40.577556862 +0000
@@ -65,6 +65,7 @@
     };
 
     // encoding / toString values
+#undef None
     enum FormattingOption {
         None = 0x0,
         RemoveScheme = 0x1,
--- qt-everywhere-opensource-src-4.6.4.org/src/gui/painting/qprinter.h	2013-10-25 12:36:39.198114736 +0000
+++ qt-everywhere-opensource-src-4.6.4/src/gui/painting/qprinter.h	2013-10-25 18:22:50.386369133 +0000
@@ -97,6 +97,9 @@
     enum PageOrder   { FirstPageFirst,
                        LastPageFirst };
 
+#undef Color
+#undef GrayScale
+
     enum ColorMode   { GrayScale,
                        Color };
 
--- qt-everywhere-opensource-src-4.6.4.org/src/corelib/io/qdir.h	2013-10-25 12:36:38.698114754 +0000
+++ qt-everywhere-opensource-src-4.6.4/src/corelib/io/qdir.h	2013-10-25 18:19:53.513375482 +0000
@@ -100,6 +100,19 @@
     typedef Filters FilterSpec;
 #endif
 
+#undef Name
+#undef Time
+#undef Size
+#undef Unsorted
+#undef SortByMask
+#undef DirsFirst
+#undef Reversed
+#undef IgnoreCase
+#undef DirsLast
+#undef LocaleAware
+#undef Type
+#undef NoSort
+
     enum SortFlag { Name        = 0x00,
                     Time        = 0x01,
                     Size        = 0x02,
--- qt-everywhere-opensource-src-4.6.4.org/src/corelib/io/qtextstream.h	2013-10-25 12:36:38.696114754 +0000
+++ qt-everywhere-opensource-src-4.6.4/src/corelib/io/qtextstream.h	2013-10-25 18:05:56.753405518 +0000
@@ -86,6 +86,7 @@
         AlignCenter,
         AlignAccountingStyle
     };
+#undef Ok
     enum Status {
         Ok,
         ReadPastEnd,
--- qt-everywhere-opensource-src-4.6.4.org/src/opengl/qglframebufferobject.cpp	2013-10-25 12:36:38.798114751 +0000
+++ qt-everywhere-opensource-src-4.6.4/src/opengl/qglframebufferobject.cpp	2013-10-25 18:29:00.921355832 +0000
@@ -39,10 +39,11 @@
 **
 ****************************************************************************/
 
+#include <qdebug.h>
+
 #include "qglframebufferobject.h"
 #include "qglframebufferobject_p.h"
 
-#include <qdebug.h>
 #include <private/qgl_p.h>
 #if !defined(QT_OPENGL_ES_1) && !defined(QT_OPENGL_ES_1_CL)
 #include <private/qpaintengineex_opengl2_p.h>
