--- qt-everywhere-opensource-src-4.6.4.org/mkspecs/linux-g++/qplatformdefs.h	2013-10-25 12:36:38.475114762 +0000
+++ qt-everywhere-opensource-src-4.6.4/mkspecs/linux-g++/qplatformdefs.h	2013-10-25 12:37:11.830113565 +0000
@@ -150,11 +150,7 @@
 #define QT_SIGNAL_ARGS		int
 #define QT_SIGNAL_IGNORE	SIG_IGN
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
 #define QT_SOCKLEN_T		socklen_t
-#else
-#define QT_SOCKLEN_T		int
-#endif
 
 #if defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)
 #define QT_SNPRINTF		::snprintf
