--- qt-everywhere-opensource-src-4.6.4.org/src/network/ssl/qsslsocket_openssl.cpp	2013-10-25 12:36:38.791114751 +0000
+++ qt-everywhere-opensource-src-4.6.4/src/network/ssl/qsslsocket_openssl.cpp	2013-10-25 12:47:06.423092222 +0000
@@ -827,11 +827,11 @@
             QString peerName = (verificationPeerName.isEmpty () ? q->peerName() : verificationPeerName);
             QString commonName = configuration.peerCertificate.subjectInfo(QSslCertificate::CommonName);
 
-            if (!isMatchingHostname(commonName.lower(), peerName.lower())) {
+            if (!isMatchingHostname(commonName.toLower(), peerName.toLower())) {
                 bool matched = false;
                 foreach (const QString &altName, configuration.peerCertificate
                          .alternateSubjectNames().values(QSsl::DnsEntry)) {
-                    if (isMatchingHostname(altName.lower(), peerName.lower())) {
+                    if (isMatchingHostname(altName.toLower(), peerName.toLower())) {
                         matched = true;
                         break;
                     }
