--- rsyslog-7.4.9-old/tools/omfile.c
+++ rsyslog-7.4.9/tools/omfile.c
@@ -48,9 +48,7 @@
 #include <libgen.h>
 #include <unistd.h>
 #include <sys/file.h>
-#ifdef OS_SOLARIS
-#	include <fcntl.h>
-#endif
+#include <fcntl.h>
 #ifdef HAVE_ATOMIC_BUILTINS
 #	include <pthread.h>
 #endif
--- rsyslog-7.4.9-old/tools/pidfile.c
+++ rsyslog-7.4.9/tools/pidfile.c
@@ -36,9 +36,7 @@
 #include <string.h>
 #include <errno.h>
 #include <signal.h>
-#ifdef __sun
 #include <fcntl.h>
-#endif
 
 #include "srUtils.h"
 
