--- screen-4.0.3.org/extern.h
+++ screen-4.0.3/extern.h
@@ -146,7 +146,7 @@
 #ifdef ZMODEM
 extern void  zmodem_abort __P((struct win *, struct display *));
 #endif
-extern void  execvpe __P((char *, char **, char **));
+extern void  screen_execvpe __P((char *, char **, char **));
 
 /* utmp.c */
 #ifdef UTMPOK
--- screen-4.0.3.org/display.c
+++ screen-4.0.3/display.c
@@ -3908,7 +3908,7 @@
       signal(SIGPIPE, SIG_DFL);
 #endif
       display = 0;
-      execvpe(*cmdv, cmdv, NewEnv + 3);
+      screen_execvpe(*cmdv, cmdv, NewEnv + 3);
       Panic(errno, *cmdv);
     default:
       break;
--- screen-4.0.3.org/window.c
+++ screen-4.0.3/window.c
@@ -1375,8 +1375,8 @@
 	proc++;
       if (!*proc)
 	proc = DefaultShell;
-      debug1("calling execvpe %s\n", proc);
-      execvpe(proc, args, NewEnv);
+      debug1("calling screen_execvpe %s\n", proc);
+      screen_execvpe(proc, args, NewEnv);
       debug1("exec error: %d\n", errno);
       Panic(errno, "Cannot exec '%s'", proc);
     default:
@@ -1388,7 +1388,7 @@
 }
 
 void
-execvpe(prog, args, env)
+screen_execvpe(prog, args, env)
 char *prog, **args, **env;
 {
   register char *path = NULL, *p;
