--- screen-4.0.3.org/comm.sh
+++ screen-4.0.3/comm.sh
@@ -18,7 +18,7 @@
 /*
  * This file is automagically created from comm.c -- DO NOT EDIT
  */
-
+#include <stdint.h>
 struct comm
 {
   char *name;
@@ -58,7 +58,7 @@
 
 struct action
 {
-  int nr;
+  intptr_t nr;
   char **args;
   int *argl;
 };
--- screen-4.0.3.org/process.c
+++ screen-4.0.3/process.c
@@ -5466,7 +5466,7 @@
       *buf = 0;
       return;
     }
-  act.nr = (int)data;
+  act.nr = (intptr_t)data;
   act.args = noargs;
   act.argl = 0;
   DoAction(&act, -1);
