--- usb-modeswitch-data-20140529.org/Makefile
+++ usb-modeswitch-data-20140529/Makefile
@@ -26,14 +26,14 @@
 	./gen-rules.tcl
 
 db-install: files-install
-	install --mode=644 -t $(PREFIX)/share/usb_modeswitch ./usb_modeswitch.d/*
+	install -D --mode=644 ./usb_modeswitch.d/* $(PREFIX)/share/usb_modeswitch 
 
 db-install-packed:
 	@# Create a compressed tar without gzip timestamp, so tar.gz
 	@# differs only if content is different
 	cd ./usb_modeswitch.d; tar -cf ../configPack.tar *
 	gzip -f9n ./configPack.tar
-	install --mode=644 -t $(PREFIX)/share/usb_modeswitch ./configPack.tar.gz
+	install --mode=644 -D ./configPack.tar.gz $(PREFIX)/share/usb_modeswitch 
 	rm -f ./configPack.tar.gz
 
 rules-reload:
