--- wine-1.5.19/dlls/ws2_32/socket.c	2012-12-07 19:17:32.000000000 +0000
+++ wine-1.5.19.patched/dlls/ws2_32/socket.c	2012-12-19 05:31:54.622000002 +0000
@@ -2791,7 +2791,7 @@
             return SOCKET_ERROR;
         } /* end switch(optname) */
     }/* end case WS_SOL_SOCKET */
-#ifdef HAVE_IPX
+#if 0
     case NSPROTO_IPX:
     {
         struct WS_sockaddr_ipx addr;
--- wine-1.5.19/dlls/ws2_32/socket.c	2012-12-19 05:34:11.083000002 +0000
+++ wine-1.5.19.patched/dlls/ws2_32/socket.c	2012-12-19 05:43:27.807000003 +0000
@@ -4318,7 +4318,7 @@
         }
         break; /* case WS_SOL_SOCKET */
 
-#ifdef HAVE_IPX
+#if 0
     case NSPROTO_IPX:
         switch(optname)
         {
