--- xmms-1.2.6.org/wmxmms/wmxmms.c	2013-01-02 22:46:18.511000002 +0000
+++ xmms-1.2.6/wmxmms/wmxmms.c	2013-01-02 23:30:37.358000003 +0000
@@ -103,8 +103,6 @@
 
 /* Does anyone know a better way? */
 
-extern Window gdk_leader_window;
-
 void action_play(void)
 {
 	xmms_remote_play(xmms_session);
@@ -635,7 +633,7 @@
 
 	pixmap = gdk_pixmap_create_from_xpm_d(window->window, &mask, NULL, xmms_dock_master_xpm);
 
-	leader = gdk_window_foreign_new(gdk_leader_window);
+	leader = gdk_window_foreign_new(gdk_x11_get_default_root_xwindow());
 	gdk_window_set_icon(leader, icon_win->window, NULL, NULL);
 	gdk_window_reparent(icon_win->window, leader, 0, 0);
 
