--- xmms2-0.8DrO_o/.waf-1.6.7-b271784c448ab906d5606267fb7f92c0/waflib/Tools/c_config.py
+++ xmms2-0.8DrO_o.patched/.waf-1.6.7-b271784c448ab906d5606267fb7f92c0/waflib/Tools/c_config.py
@@ -164,6 +164,11 @@
 		return
 	if'args'in kw:
 		lst+=Utils.to_list(kw['args'])
+
+
+	lst.append(lst.pop(1))
+
+
 	ret=self.cmd_and_log(lst)
 	if not'okmsg'in kw:
 		kw['okmsg']='yes'
@@ -686,4 +691,4 @@
 conf(cxx_load_tools)
 conf(get_cc_version)
 conf(add_as_needed)
-conf(multicheck)
\ No newline at end of file
+conf(multicheck)
